drgn/helpers/linux/mm.py

```diff
@@ -1523,3 +1526,16 @@ def get_task_rss_info(prog: Program, task: Object) -> TaskRss:
                 shmemrss += gtask.rss_stat.count[MM_SHMEMPAGES].value_()
 
     return TaskRss(filerss, anonrss, shmemrss, swapents)
+
+
+def get_task_vmem(task: Object) -> float:
+    """
+    Return virtual memory size of the task
+    """
+    prog = task.prog_
+    page_size = prog["PAGE_SIZE"].value_()
+    try:
+        vmem = (task.mm.total_vm.value_() * page_size) // ByteToKB
+    except drgn.FaultError:
+        vmem = 0
+    return vmem
```

Comment:
A few comments on this one:

* We can probably name this after [`task_vsize()`](https://github.com/torvalds/linux/blob/347e9f5043c89695b01e66b3ed111755afcf1911/fs/proc/task_mmu.c#L86-L89).
* Kilobytes are a pretty arbitrary unit to return from a helper. Let's just return bytes (like `task_vsize()` does, and also delete the definition of `ByteToKB`).
* Also, if the intention of the `FaultError` handling is to handle kernel threads with a `NULL` `mm`, it'd make more sense to check for that explicitly instead.

Reply:
I feel like I remember searching for a constant like this and not finding it!

I'm not against switching over. I do feel like it's easier to change a task constant than it is to change behavior, so there's an argument to be made that this implementation is less likely to become incorrect over time.

But I guess there's nothing that says that `task_state_to_char(t)` for a kthread _couldn't_ return Z at some point in the future, it's not like that's an invariant specified anywhere in the kernel.